### ISSUE TITLE
tests: net: lib: lwm2m: content_json: net.lwm2m.content_json fails

### ISSUE THREAD

**Issue by katgiadla** [zephyrproject-rtos/zephyr#62329]
**Describe the bug**
The test `net.lwm2m.content_json` fails.

Observed for `nrf52840dk_nrf52840`.

**To Reproduce**

1. have `nrf52840dk_nrf52840` connected
2. go to your zephyr dir
3. call `./scripts/twister -T tests/net/lib/lwm2m/content_json -p nrf52840dk_nrf52840 --device-testing --device-serial /dev/ttyACM0 -v --inline-logs`
4. See console output with error

**Expected behavior**
Valid console output

**Impact**
Not clear

**Logs and console output**
```
*** Booting Zephyr OS build zephyr-v3.4.0-3264-g5ba4288fea6d ***
Running TESTSUITE net_content_json
===================================================================
START - test_get_bool
PASS - test_get_bool in 0.001 seconds
===================================================================
START - test_get_float
PASS - test_get_float in 0.002 seconds
===================================================================
START - test_get_objlnk
PASS - test_get_objlnk in 0.001 seconds
===================================================================
START - test_get_s32
PASS - test_get_s32 in 0.001 seconds
===================================================================
START - test_get_s64
PASS - test_get_s64 in 0.001 seconds
===================================================================
START - test_get_string
PASS - test_get_string in 0.001 seconds
===================================================================
START - test_put_bool
PASS - test_put_bool in 0.001 seconds
===================================================================
START - test_put_float
Assertion failed at ../../../../../../../../../../zephyr/zephyr/tests/net/lib/lwm2m/content_json/src/main.c:381: net_content_json_test_put_float: (test_msg.msg_data + offset not equal to expected_payload[i])
Invalid payload format
FAIL - test_put_float in 0.022 seconds
===================================================================
START - test_put_objlnk
PASS - test_put_objlnk in 0.001 seconds
===================================================================
START - test_put_s16
PASS - test_put_s16 in 0.001 seconds
===================================================================
START - test_put_s32
PASS - test_put_s32 in 0.001 seconds
===================================================================
START - test_put_s64
Assertion failed at ../../../../../../../../../../zephyr/zephyr/tests/net/lib/lwm2m/content_json/src/main.c:305: net_content_json_test_put_s64: (test_msg.msg_data + offset not equal to expected_payload[i])
Invalid payload format
FAIL - test_put_s64 in 0.021 seconds
===================================================================
START - test_put_s8
PASS - test_put_s8 in 0.001 seconds
===================================================================
START - test_put_string
PASS - test_put_string in 0.001 seconds
===================================================================
TESTSUITE net_content_json failed.
Running TESTSUITE net_content_json_nodata
===================================================================
E: no more instances available for obj 65535
START - test_get_bool_nodata
E: JSON object bits not valid -22
PASS - test_get_bool_nodata in 0.004 seconds
===================================================================
START - test_get_float_nodata
E: JSON object bits not valid -22
PASS - test_get_float_nodata in 0.004 seconds
===================================================================
START - test_get_objlnk_nodata
E: JSON object bits not valid -22
PASS - test_get_objlnk_nodata in 0.004 seconds
===================================================================
START - test_get_s64_nodata
E: JSON object bits not valid -22
PASS - test_get_s64_nodata in 0.004 seconds
===================================================================
START - test_get_string_nodata
E: JSON object bits not valid -22
PASS - test_get_string_nodata in 0.004 seconds
===================================================================
TESTSUITE net_content_json_nodata succeeded
Running TESTSUITE net_content_json_nomem
===================================================================
E: no more instances available for obj 65535
START - test_get_s32_nodata
E: JSON object bits not valid -22
PASS - test_get_s32_nodata in 0.004 seconds
===================================================================
START - test_put_bool_nomem
PASS - test_put_bool_nomem in 0.001 seconds
===================================================================
START - test_put_float_nomem
PASS - test_put_float_nomem in 0.001 seconds
===================================================================
START - test_put_objlnk_nomem
PASS - test_put_objlnk_nomem in 0.001 seconds
===================================================================
START - test_put_s16_nomem
PASS - test_put_s16_nomem in 0.001 seconds
===================================================================
START - test_put_s32_nomem
PASS - test_put_s32_nomem in 0.001 seconds
===================================================================
START - test_put_s64_nomem
PASS - test_put_s64_nomem in 0.001 seconds
===================================================================
START - test_put_s8_nomem
PASS - test_put_s8_nomem in 0.001 seconds
===================================================================
START - test_put_string_nomem
PASS - test_put_string_nomem in 0.001 seconds
===================================================================
TESTSUITE net_content_json_nomem succeeded
------ TESTSUITE SUMMARY START ------
SUITE FAIL -  85.71% [net_content_json]: pass = 12, fail = 2, skip = 0, total = 14 duration = 0.056 seconds
- PASS - [net_content_json.test_get_bool] duration = 0.001 seconds
- PASS - [net_content_json.test_get_float] duration = 0.002 seconds
- PASS - [net_content_json.test_get_objlnk] duration = 0.001 seconds
- PASS - [net_content_json.test_get_s32] duration = 0.001 seconds
- PASS - [net_content_json.test_get_s64] duration = 0.001 seconds
- PASS - [net_content_json.test_get_string] duration = 0.001 seconds
- PASS - [net_content_json.test_put_bool] duration = 0.001 seconds
- FAIL - [net_content_json.test_put_float] duration = 0.022 seconds
- PASS - [net_content_json.test_put_objlnk] duration = 0.001 seconds
- PASS - [net_content_json.test_put_s16] duration = 0.001 seconds
- PASS - [net_content_json.test_put_s32] duration = 0.001 seconds
- FAIL - [net_content_json.test_put_s64] duration = 0.021 seconds
- PASS - [net_content_json.test_put_s8] duration = 0.001 seconds
- PASS - [net_content_json.test_put_string] duration = 0.001 seconds
SUITE PASS - 100.00% [net_content_json_nodata]: pass = 5, fail = 0, skip = 0, total = 5 duration = 0.020 seconds
- PASS - [net_content_json_nodata.test_get_bool_nodata] duration = 0.004 seconds
- PASS - [net_content_json_nodata.test_get_float_nodata] duration = 0.004 seconds
- PASS - [net_content_json_nodata.test_get_objlnk_nodata] duration = 0.004 seconds
- PASS - [net_content_json_nodata.test_get_s64_nodata] duration = 0.004 seconds
- PASS - [net_content_json_nodata.test_get_string_nodata] duration = 0.004 seconds
SUITE PASS - 100.00% [net_content_json_nomem]: pass = 9, fail = 0, skip = 0, total = 9 duration = 0.012 seconds
- PASS - [net_content_json_nomem.test_get_s32_nodata] duration = 0.004 seconds
- PASS - [net_content_json_nomem.test_put_bool_nomem] duration = 0.001 seconds
- PASS - [net_content_json_nomem.test_put_float_nomem] duration = 0.001 seconds
- PASS - [net_content_json_nomem.test_put_objlnk_nomem] duration = 0.001 seconds
- PASS - [net_content_json_nomem.test_put_s16_nomem] duration = 0.001 seconds
- PASS - [net_content_json_nomem.test_put_s32_nomem] duration = 0.001 seconds
- PASS - [net_content_json_nomem.test_put_s64_nomem] duration = 0.001 seconds
- PASS - [net_content_json_nomem.test_put_s8_nomem] duration = 0.001 seconds
- PASS - [net_content_json_nomem.test_put_string_nomem] duration = 0.001 seconds
------ TESTSUITE SUMMARY END ------
===================================================================
RunID: cbe95c8d0cb46e0e93561358ad163d45
PROJECT EXECUTION FAILED
```

**Environment (please complete the following information):**

- OS: Ubuntu 20.04.1 LTS
- Toolchain Zephyr SDK 0.16.0
- Commit SHA or Version used: zephyr-v3.4.0-3264-g5ba4288fea6d

**Comment by katgiadla** [comment]
The same issue occurred for `nrf5340dk_nrf5340_cpunet`.

**Comment by katgiadla** [comment]
It is fixed like in #62330.
